tracing/tests/support/subscriber.rs

```diff
@@ -43,21 +43,21 @@ struct Running<F: Fn(&Metadata<'_>) -> bool> {
     filter: F,
 }
 
-pub struct MockSubscriber<F: Fn(&Metadata<'_>) -> bool> {
+pub struct SubscriberTest<F: Fn(&Metadata<'_>) -> bool> {
```

Comment:
WDYT of `TestSubscriber` instead?

Reply:
SGTM

Reply:
Actually, iunno, calling it TestSubscriber seems to imply that its actually a subscriber, but its not right? SubscriberTest seems clearer to me, in that we're setting up a Test of a Subscriber, and the object itself is first and foremost a representation of the Test being run.